### DISCUSSION TITLE
how to upgrade magento 247 to 247 p2 using docker in ubuntu

### DISCUSSION THREAD

**Discussion by bhushanmeetanshi** [markshust/docker-magento#1228] [General]
how to upgrade magento 247 to 247 p2 using docker in ubuntu

**Comment by bhushanmeetanshi** [comment]
@YevhenZvieriev do you have any idea about this Question?

**Reply by YevhenZvieriev** [reply]
Hi, @bhushanmeetanshi 

I'll help you, but I'd like to clarify something.

Do you want to upgrade a commercial project or your local sandbox Magento 2 instance?

**Comment by YevhenZvieriev** [comment]
To upgrade Magento from 2.4.7 to 2.4.7-p2, follow these steps:

### 1. Create a backup copy

Before performing any updates, I recommend making a backup of the database and store files.

### 2.  Switch the site to maintenance mode

Enable maintenance mode

```
bin/magento maintenance:enable
```

## 3. Update Magento through Composer
Update Magento dependencies via Composer to version 2.4.7-p2

```
bin/composer require magento/product-community-edition=2.4.7-p2 --no-update
``` 

If you got error, then you should execute this command

```
bin/composer require-commerce magento/product-community-edition=2.4.7-p2 --no-update
``` 

Update your dependencies:

```
bin/composer update
```

## 4. Clear the cache and restart the compilation

After upgrading Composer, you need to run the script to refresh the database and clear the cache:

```
bin/deploy
```

## 5. Check if the store is working correctly after the update

To confirm that you've updated successfully, run the following command:

```
bin/magento -V
```

This will show you the current Magento version:

![Screenshot from 2024-09-26 14-03-31](https://github.com/user-attachments/assets/ca9c6bec-4220-430d-89b7-2833cad0ca2d)


If you have third-party modules, please make sure they are compatible with version 2.4.7-p2.

**Reply by bhushanmeetanshi** [reply]
I have local project.

**Reply by bhushanmeetanshi** [reply]
I will try this steps as you mentioned. thanks for your kind reply.
